.github/workflows/automated_release.yml

```diff
@@ -0,0 +1,43 @@
+# Automatically generate a release on develop and master
+name: Automated release and build of project
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set release tag
+      # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+      run: >-
+        echo "RELEASE_VERSION=$([ ${GITHUB_REF##*/} = "develop" ] && printf '%s\n' $(cat .release-version)-develop || printf '%s\n' $(cat .release-version))" >> $GITHUB_ENV
+
+    - name: Setup node
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Install
+      run: >-
+        yarn install
+
+    - name: Compile Build
+      run: >-
+        ./compile-build.sh release.tar.gz dist
+
+    - name: Create release and upload release.tar.gz
+      uses: softprops/action-gh-release@v1
+      with:
+        name: ${{ env.RELEASE_VERSION }}
+        tag_name: v${{ env.RELEASE_VERSION }}
+        prerelease: ${{ !(github.ref == 'refs/heads/master') }}
+        files: release.tar.gz
```

Comment:
don't know much about this action but might be worth double checking the 'files param' wants the file name and not the file path as path would be ./release.tar.gz

Reply:
I think this should work as is.